### PR TITLE
Use a custom vars file instead of patching

### PR DESCRIPTION
**Pull request by zmc**
Instead of maintaining a patch for var.tfvars and various sed
invocations, we can use a second, custom vars file (site.tfvars) and
populate its values with envsubst.

Signed-off-by: Zack Cerza <zack@redhat.com>

### CODE REVIEW

**Review by lukebrowning — CHANGES_REQUESTED** [review]
The ocp4-upi-kvm.patch file has a one line change outside of the vars.tfvar file.  This change should be preserved.   It may not be needed in the long run, but I have run into problems where the wrong ignition file is being rendered in the bastion node.   We need to debug this issue before we completely eliminate this file.

File: scripts/helper/create-cluster.sh [hunk 1]
```diff
@@ -276,48 +283,20 @@ case "$OCP_VERSION" in
 4.4|4.5)
 	patch -p1 < $WORKSPACE/ocs-upi-kvm/files/ocp4-upi-kvm.legacy.patch
```
Comment (lukebrowning) [inline scripts/helper/create-cluster.sh:284]: Let's make this conditional on the presence of the patch.   There is now release-4.5 branch that we can use, except it doesn't have all of the fixes that we need.  I have been advocating for fixes to be cherry picked to this release.

File: scripts/helper/create-cluster.sh [hunk 2]
```diff
@@ -276,48 +283,20 @@ case "$OCP_VERSION" in
...
 	;;
-*)
-	patch -p1 < $WORKSPACE/ocs-upi-kvm/files/ocp4-upi-kvm.patch
```
Comment (lukebrowning) [inline scripts/helper/create-cluster.sh:280]: Let's make this conditional on the presence of the patch also.   We may need to develop and test fixes to this release.

**Comment by zmc** [comment]
I must have missed the second chunk of that patch - looks like it was added last night. I'll reintroduce that.

File: scripts/helper/create-cluster.sh [hunk 3]
```diff
@@ -23,12 +23,18 @@ if [ ! -e helper/parameters.sh ]; then
 fi
 
 source helper/parameters.sh
+export SANITIZED_OCP_VERSION
+
+function terraform_apply () {
+  cat $WORKSPACE/ocs-upi-kvm/files/site.tfvars.in | envsubst > $WORKSPACE/site.tfvars
```
Comment (lukebrowning) [inline scripts/helper/create-cluster.sh:29]: Please follow the indenting scheme of the file -- 8 space tab.   See below also...

**Comment by zmc** [comment]
Patch is restored and the indentation is fixed. Re: conditional patching - let's do that in a separate PR as this one no longer needs to touch that code.

**Review by lukebrowning — APPROVED** [review]

**Pull request merged**